### PR TITLE
Core: Add support HikariCP

### PR DESCRIPTION
**Pull request by marki1an**

### CODE REVIEW

**Review by Net-burst — DISMISSED** [review]
Generally LGTM, apart from 30sec default c3p0 acquisition time. That one looks too big. Let's halve that.

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]
Pls add to project properties default value as c3p0

File: src/main/java/org/prebid/server/spring/config/SettingsConfiguration.java [hunk 1]
```diff
@@ -134,21 +134,25 @@ private static BasicJdbcClient createBasicJdbcClient(
 
         @Bean
         JDBCClient vertxJdbcClient(Vertx vertx, StoredRequestsDatabaseProperties storedRequestsDatabaseProperties) {
+            final boolean isHikari = storedRequestsDatabaseProperties.getProviderClass() == DbPoolType.hikari;
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/spring/config/SettingsConfiguration.java:137]: Use switch for properties resolving, can be useful when we will have more than two values

File: pom.xml [hunk 2]
```diff
@@ -357,6 +358,11 @@
             <artifactId>log4j-api</artifactId>
             <version>${log4j.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.zaxxer</groupId>
+            <artifactId>HikariCP</artifactId>
+            <version>5.0.1</version>
```
Comment (SerhiiNahornyi) [inline pom.xml:364]: Add version to properties section

File: src/test/java/org/prebid/server/settings/JdbcApplicationSettingsTest.java [hunk 3]
```diff
@@ -661,6 +661,8 @@ private JdbcClient jdbcClient() {
                 new JsonObject()
                         .put("url", JDBC_URL)
                         .put("driver_class", "org.h2.Driver")
-                        .put("max_pool_size", 10)), metrics, clock);
+                        .put("max_pool_size", 10)
+                        .put("provider_class", "io.vertx.ext.jdbc.spi.impl.C3P0DataSourceProvider")),
```
Comment (SerhiiNahornyi) [inline src/test/java/org/prebid/server/settings/JdbcApplicationSettingsTest.java:665]: replace url on 662 line with jdbcUrl and 665 with hikari type

Comment (marki1an) [reply src/test/java/org/prebid/server/settings/JdbcApplicationSettingsTest.java:665]: it doesn't need when we use c3p0

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**